src/fieldExtensions.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * The flattenState() field extension allows the user to access fields normally like
+ * `state.field` instead of `state.present.field`.
+ *
+ * Warning: if your state has fields that conflict with redux-undo's like .past or .index
+ * they will be overridden. You must access them as `state.present.past` or `state.present.index`
+ */
+
+export const flattenState = () => {
+  return (undoableConfig) => {
+    console.log(undoableConfig)
+    if (!undoableConfig.disableWarnings) {
+      console.warn(
```

Comment:
maybe we should only show this warning if we detect that the state contains one of redux-undo's field names, or an array/primitive?

Reply:
I thought about that. The issue is you would be checking the returned object for *every* action that was dispatched to the wrapped reducer (there is no way to check only the first time and then trust it from then on).

I thought it would be best to warn the user exactly once at initialization, no matter what the user has in their state. It should also be very clear in the documentation that this is the case (I can make it more explicit).

Thoughts?

Reply:
yeah, I see, that makes sense. then let's keep it as is but put the `disableWarnings` option as an argument to the function. let's also separate the field extensions into a separate package.

Reply:
Sounds good :ok_hand: